purchase_order_analytic_search/tests/test_analytic_search.py

```diff
@@ -0,0 +1,173 @@
+                        "account_analytic_id": cls.analytic_account_2.id,
+                    },
+                ),
+            ],
+        }
+        cls.purchase_order_2 = cls.purchase_order_model.create(po_dict2)
+        # PURCHASE ORDER NUM 3 => account 2
+        po_dict3 = {
+            "partner_id": cls.partner2.id,
+            "order_line": [
+                (
+                    0,
+                    0,
+                    {
+                        "date_planned": Datetime.now(),
+                        "name": "PO01",
+                        "product_id": cls.product.id,
+                        "product_uom": uom_id,
+                        "price_unit": 1,
+                        "product_qty": 5.0,
+                        "account_analytic_id": cls.analytic_account_2.id,
+                    },
+                ),
+                (
+                    0,
+                    0,
+                    {
+                        "date_planned": Datetime.now(),
+                        "name": "PO01",
+                        "product_id": cls.product.id,
+                        "product_uom": uom_id,
+                        "price_unit": 1,
+                        "product_qty": 5.0,
+                        "account_analytic_id": cls.analytic_account_2.id,
+                    },
+                ),
+            ],
+        }
+        cls.purchase_order_3 = cls.purchase_order_model.create(po_dict3)
+
+    def test_filter_analytic_accounts(self):
+        found = self.purchase_order_model.search(
+            [("account_analytic_ids", "=", self.analytic_account_1.id)]
+        )
+        self.assertEqual(found, self.purchase_order_1 + self.purchase_order_2)
+
+    def test_filter_analytic_accounts_by_name(self):
+        found = self.purchase_order_model.search(
+            [("account_analytic_ids", "=", "nt 1")]
+        )
+        self.assertEqual(found, self.purchase_order_1 + self.purchase_order_2)
+
+    def test_filter_analytic_accounts_not_equal(self):
+        found = self.purchase_order_model.search(
+            [("account_analytic_ids", "!=", self.analytic_account_1.id)]
+        )
+        self.assertTrue(self.purchase_order_3 in found)
+        self.assertTrue(self.purchase_order_1 not in found)
+        self.assertTrue(self.purchase_order_2 not in found)
+
+    def test_compute_analytic_accounts(self):
+        self.assertEqual(
+            self.purchase_order_1.account_analytic_ids, self.analytic_account_1
+        )
+        self.assertEqual(
+            self.purchase_order_2.account_analytic_ids,
+            self.analytic_account_1 + self.analytic_account_2,
+        )
+
+    def test_filter_analytic_accounts_equal_false(self):
+        found = self.purchase_order_model.search([("account_analytic_ids", "=", False)])
+        self.assertFalse(self.purchase_order_1 in found)
+        self.assertFalse(self.purchase_order_2 in found)
+        self.assertFalse(self.purchase_order_3 in found)
```

Comment:
```suggestion
        self.assertFalse(found)
```
By the coding logic, if no value, result will be nothing. So, one test line is enough.